components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java

```diff
@@ -263,28 +263,25 @@ public int getMaxLoginAttemptCount() {
 
     private String buildUrl(String defaultContext, Supplier<String> getValueFromFileBasedConfig) {
 
-        String url;
         if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
-            url = buildTenantQualifiedUrl(defaultContext);
+            return buildAbsoluteUrl(defaultContext);
         } else {
-            String authenticationEndpointURLInFile = getValueFromFileBasedConfig.get();
-            if (StringUtils.isNotBlank(authenticationEndpointURLInFile)) {
-                // Use the value configured in the file.
-                url = authenticationEndpointURLInFile;
+            String urlFromFileBasedConfig = getValueFromFileBasedConfig.get();
+            if (StringUtils.isNotBlank(urlFromFileBasedConfig)) {
+                // If the file based URL is set, then we have to return the file based URL.
+                return urlFromFileBasedConfig;
             } else {
-                // Use the default context.
-                url = defaultContext;
+                return defaultContext;
             }
         }
-        return url;
     }
 
-    private String buildTenantQualifiedUrl(String context) {
+    private String buildAbsoluteUrl(String context) {
 
         try {
             return ServiceURLBuilder.create().addPath(context).build().getAbsolutePublicURL();
         } catch (URLBuilderException e) {
-            throw new IdentityRuntimeException("Error while building url for context: " + context);
+            throw new IdentityRuntimeException("Error while building tenant qualified url for context: " + context);
```

Comment:
I would prefer if this exception can be thrown and handled at the public API layer